src/pages/guides/tutorials/stats-addon.md

```diff
@@ -0,0 +1,1093 @@
+addOnUISdk.ready.then(async () => {
+  // API to expose to the Document Sandbox
+  const iframeApi = { 
+    createTable(documentData) { /* ... */ },
+    toggleStatus(sdk) { /* ... */ }
+  };
+  runtime.exposeApi(iframeApi);
+  
+  // Toggle the iframe SDK status light
+  iframeApi.toggleStatus("iframe");
+  
+  // Import the Document Sandbox API proxy
+  const { runtime } = addOnUISdk.instance;
+  const sandboxProxy = await runtime.apiProxy("documentSandbox");
+  // Exposing the iFrame API to the Document Sandbox.
+  
+  // Set the button's click handler
+  const statsButton = document.getElementById("stats");
+  statsButton.addEventListener("click", async () => {
+    // Invoke the Document API via Document Sandbox proxy
+    // to get the document metadata and initiate the table creation process
+    await sandboxProxy.getDocumentData(); // 👈 mind the await
+  });
+
+  // Enable the button only when the addOnUISdk is ready
+  statsButton.disabled = false;
+});
+```
+
+Besides the usual SWC imports, everything must be wrapped by a callback invoked when the `addOnUISdk` module is `ready`—which also means we can switch the first SDK status light to green. Please note that `toggleStatus()` must be available to both the iframe and the Document Sandbox: declaring the `iframeApi` constant first (lines 18-21) and passing it to `exposeApi()` later allows me to call it (line 21).
+
+```js
+// ❌ _in this case_ it's the wrong syntax choice!
+runtime.exposeApi({ 
+  createTable(documentData) { /* ... */ },
+  toggleStatus(sdk) { /* ... */ }
+});
+
+// ✅ better for our needs _in this context_
+const iframeApi = { 
+  createTable(documentData) { /* ... */ },
+  toggleStatus(sdk) { /* ... */ }
+});
+runtime.exposeApi(iframeApi);
+// toggleStatus is accessible in the iframe too
+iframeApi.toggleStatus("iframe"); // 👈
+```
+
+<InlineAlert variant="info" slots="text" />
+
+Let me remind you again of the need to `await` when invoking `getDocumentData()` (line 37, `index.js`), as the Communication API wraps proxy calls with Promises.
+
+In `documentSandbox/code.js`, we bring the iframe proxy in, toggle the status light, and expose the `getDocumentData()` function. Please note that it must be declared asynchronous (line 7) because of the need to `await` when invoking the `panelUIProxy` method `createTable()` (line 12).
+
+<CodeBlock slots="heading, code" repeat="1" languages="code.js"/>
+
+#### documentSandbox/code.js
+
+```js
+import addOnSandboxSdk from "add-on-sdk-document-sandbox";
+const { runtime } = addOnSandboxSdk.instance;
+
+async function start() {
+  const panelUIProxy = await runtime.apiProxy("panel");
+  runtime.exposeApi({
+    async getDocumentData() {                    // 👈 async keyword, because 👇
+      // Get the document's metadata
+      let documentData;
+      // ... TODO
+      // ... then, directly invoke the iframe method
+      await panelUIProxy.createTable(documentData); // 👈 the Communication API is async 👆
```

Comment:
Although this is a valid syntax and demonstrate the calling of the iframe exosed apis from script runtime - 
in the context of this example - this might be confusing and might give an impression to the reader that _this is how_ the results / data needs to be communicated back to the caller.
note that we can just return the required data from this method and let the iframe ui decide for itself whether to create a table from the given data or something else ( based on the current selection criteria in the UI ) - that way, scritp backend becomes a pure model and provides the document data while the iframe would become the consume of the data - which is what i am assuming most of the use-cases will be like and this example would help demostrate the working of the same.

Reply:
You're right, I'll add a box to clarify that this circular call is mostly for demonstration purposes and the other workflow may be more common